Resources/doc/i18n.rst

```diff
@@ -1,74 +1,58 @@
-        twig:
-            template: AppBundle:Default:knp_menu.html.twig
-
-Alternative B: I18n only for certain menus
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-If you want to translate a certain menu, add the ``template`` parameter to
-its ``knp_menu_render`` call.
+    $menu = $factory->createItem('root');
 
-.. code-block:: html+jinja
+    $menu->addChild('Home', array('route' => 'homepage'));
+    $menu->addChild('Login', array('route' => 'login'));
 
-    {{ knp_menu_render('name_of_your_menu', {'template': 'AppBundle:Default:knp_menu.html.twig'}) }}
+The items "Home" and "Login" can now be translated in the message domain:
 
-That's it.
+.. configuration-block::
 
-Add translation domain information in the controller
-----------------------------------------------------
+    .. code-block:: yaml
 
-In your menu Builder class you need to add extras. For example, imagine you have
-two menu items Home and Login.
+        # app/Resources/translations/messages.fr.yml
+        Home: Accueil
+        Login: Connexion
 
-.. code-block:: php
+    .. code-block:: xliff
 
-    $menu = $factory->createItem('root');
+        <!-- app/Resources/translations/messages.fr.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="menu.home">
+                        <source>Home</source>
+                        <target>Accueil</target>
+                    </trans-unit>
 
-    $menu->addChild('Home', array('route' => 'homepage'));
-    $menu->addChild('Login', array('route' => 'login'));
+                    <trans-unit id="menu.login">
+                        <source>Login</source>
+                        <target>Connexion</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
 
-To translate them you need to add translation_domain parameter using setExtra(s):
+    .. code-block:: php
 
-.. code-block:: php
+        // app/Resources/translations/messages.fr.php
+        return array(
+            'Home' => 'Accueil',
+            'Login' => 'Connexion',
+        );
 
-    $menu = $factory->createItem('root');
+Configure the Translation Domain
+--------------------------------
 
-    // will look for "Home" in AppBundle/Resources/translations/AppBundle.locale.yml
-    $menu->addChild('Home', array('route' => 'homepage'))->setExtra('translation_domain', 'AppBundle');
+You can configure the translation domain that's used in the extras of the menu
+item::
 
-    // will look for "Login" in Acme/AdminBundle/Resources/translations/AcmeAdminBundle.locale.yml
-    $menu->addChild('Login', array('route' => 'login'))->setExtra('translation_domain', 'AcmeAdminBundle');
+    // ...
+    $menu->addChild('Home', array('route' => 'homepage'))
+        ->setExtra('translation_domain' => 'AcmeAdminBundle');
```

Comment:
if we add the menu defaults, this domain could also be configured via the configuration, right? and should it be AppBundle rather than AcmeAdminBundle?